modules/vector-search/pages/fine-tune-vector-search.adoc

```diff
@@ -0,0 +1,132 @@
+which balances between too few and too many clusters, ensuring efficient indexing.
+
+| stem:["nvec" lt 1000]
+| N/A
+| For number of vectors less than 1000, Couchbase will carry out a straight forward one-to-one mapping between IDs and vectors with an exact vector comparison.
+Vectors are directly stored without the need for additional processing for the `nlist` calculation.
+
+|===
+
+`nprobes` (or `probes`)::
+The number of `centroids` that a query will check for similar vectors.
+The `nprobe` value is only set when Couchbase is using an Inverted File Index. Couchbase will select the best index type and comparison method depending on the size of the dataset and the `vector_index_optimized_for` setting.
++
++
+[%header, cols="3*a"]
+|===
+| Query optimization
+| `nprobe` calculation
+| Notes
+
+
+| Default calculation
+| stem:[sqrt("nlist")]
+| This provides a balanced tradeoff between recall and latency by adjusting the number of clusters probed during queries.
+
+| Latency-optimized (`vector_index_optimized_for: latency`)
+| stem:[sqrt("nlist") / 2]
+| A minimum value of 1 is enforced to avoid setting `nprobe` too low.
+
+|===
+
+
+== Fine-Tuning Query Parameters
+
+
+The Vector Search API will accept an additional parameter object, allowing for tuning the recall and/or accuracy of the search execution.
+
+`ivf_nprobe_pct`::
+Adjusts the number percentage of clusters searched during queries, allowing for fine-tuning of the balance between recall and performance.
+If the value of `nprobe` is 5% of `nlist` (the centroid count), then setting the value of `ivf_nprobe_pct` higher than 5% will have the search cover a higher percentage of clusters, which will improve the accuracy of the search.
+
+`ivf_max_codes_pct`::
+The default value for this is 100, and the value represents the percentage of `centroids` that will be visited during a search. Reducing the value reduces the number of centroids visited, which will descrease accurary and recal, but will result in faster compute times.
+
+For example, given a cluster `vector_index_optimized_for`: "recall" and `indexPartitions`: 5,  `nlist` and `nprobe` are determined based on the current vector count in a given partition:
+
+'''
+[options="noheader", frame="none", grid="none" cols="1,1"]
+|===
+| stem:["Total vectors in index"] (optimization = recall)
+| 10,000,000
+
+| stem:["Average vectors in a partition for 5 partitions total"]
+| 2,000,000
+
+| stem:["nlist"] (or stem:["centroids"]) = stem:[4 times sqrt("total vectors in index")]
+| 5657
+
+| stem:["nprobes"] = stem:[sqrt(nlist)]
+| 75
+
+| stem:["Calculated default: ivf_nprobe_pct"]
+| 1.325%
+
+| stem:["Calculated default: ivf_max_codes_pct"]
+| 100%
+
+|===
+
+'''
+
+[source, json]
+.Using tuning parameters
+----
+{
+  "fields": ["*"],
+  "knn": [{
+    "k": 10,
+    "params": {
+      "ivf_nprobe_pct": 1,
+      "ivf_max_codes_pct": 0.2
+    },
+    "field": "embedding",
+    "vector": [0.024901132253900747, 1535]
+  }]
+}
+----
```

Comment:
How does this sample code relate to what's in the table above? The values provided are completely different from what you show in the calculation. What is this doing?